src/include/machnet_engine.h

```diff
@@ -874,7 +962,73 @@ class MachnetEngine {
         if (listeners_.find(local_ipv4_addr) != listeners_.end()) {
           // We have a listener on this port.
           const auto &listeners_on_ip = listeners_[local_ipv4_addr];
+          const int qid =
+              shared_state_->GetQueueId(local_ipv4_addr, local_udp_port);
           if (listeners_on_ip.find(local_udp_port) == listeners_on_ip.end()) {
+            if (qid > -1) {
```

Comment:
Please create a new private method for this; it's significant enough so that it can be abstracted in a new method.

`const int qid` from line 965 should also go in the new method.

You can name the new method `bool MaybeRequestRetryForRSS()`.

If there is a listener and you've sent a kRetry you can return `true`, otherwise `false`.

Doing a :
`if (MaybeRequestRetryForRSS()) { LOG(INFO) << "Engine " << q_id << " requesting redirect to engine: " << new_q_id; return; }`

The Dropping packet error message at line 1032 shouldn't be printed.